api/app/controllers/api/v1/applications_controller.rb

```diff
@@ -0,0 +1,77 @@
+# frozen_string_literal: true
+
+module Api::V1
+    # Controller for Applications
+    class ApplicationsController < ApplicationController
+
+        # GET /applications
+        def index
+            if not params.include?(:session_id)
+                render json: { status: 'success', message: '', payload: Application.order(:id) }
+                return
+            end
+            if Session.exists?(id: params[:session_id])
+                render json: { status: 'success', message: '', payload: applications_by_session }
+            else
+                render json: { status: 'error', message: 'Invalid session_id', payload: {} }
+            end
+        end
+
+        # POST /applications
+        def create
+            if not Session.exists?(id: params[:session_id])
+                render json: { status: 'error', 
+                    message: 'Invalid session_id', payload: applications_by_session }
+                return
+            end
+            application = Application.new(application_params)
+            if not application.save # does not pass Application model validation
+                render json: { status: 'error', message: application.errors, payload: {} }
+            end
+            params[:application_id] = application[:id]
+            message = valid_matching
```

Comment:
This is a confusing name...can you come up with something better?

Reply:
Is `valid_applicant_matching_data` a good alternative? Or should I come up with something else?